src/entities/nestedExit/index.ts

```diff
@@ -0,0 +1,127 @@
+import { encodeFunctionData } from 'viem';
+import { Address, Hex } from '../../types';
+import { BALANCER_RELAYER, getPoolAddress } from '../../utils';
+import { Relayer } from '../relayer';
+import { TokenAmount } from '../tokenAmount';
+import { balancerRelayerAbi, bathcRelayerLibraryAbi } from '../../abi';
+import {
+    NestedExitInput,
+    NestedExitQueryResult,
+    NestedExitCallInput,
+} from './types';
+import { NestedPoolState } from '../types';
+import { doQueryNestedExit } from './doQueryNestedExit';
+import { getNestedExitCalls } from './getNestedExitCalls';
+import { parseNestedExitCall } from './parseNestedExitCall';
+import { getPeekCalls } from './getPeekCalls';
+
+export class NestedExit {
+    async query(
+        input: NestedExitInput,
```

Comment:
Could we make the `NestedExitInput` a Union of the two exit types, Proportional and SingleToken.